### PR TITLE
[snapshot] Split xcodebuild logs by device name and language

### PR DESCRIPTION
**Pull request by KrauseFx**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Description
Updated the `xcodebuild_log_file` method to include the device name, language and locale where required. This now means that there is a log file generated per run to prevent logs being overwritten.

#### Before
```
➜  ls -l ~/Library/Logs/snapshot 
-rw-r--r--  1 liamnichols  staff  25912  1 Apr 15:54 ExampleApp-ExampleScheme.log
```

#### After
```
➜  ls -l ~/Library/Logs/snapshot 
-rw-r--r--  1 liamnichols  staff  25912  1 Apr 15:54 ExampleApp-ExampleScheme-iPhone 6-en-US.log
-rw-r--r--  1 liamnichols  staff  25912  1 Apr 15:54 ExampleApp-ExampleScheme-iPhone 6-pt-pt_BR.log
-rw-r--r--  1 liamnichols  staff  25912  1 Apr 15:54 ExampleApp-ExampleScheme-iPhone SE-en-US.log
-rw-r--r--  1 liamnichols  staff  25912  1 Apr 15:54 ExampleApp-ExampleScheme-iPhone SE-pt-pt_BR.log
```

### Motivation and Context
I am currently running snapshot on an array of devices however my ui tests are failing randomly in some runs but not always the final device run. I try to look at the console output from xcpretty however it's not actually giving me anything useful. 

Instead I've tried looking at the output in `~/Library/Logs/snapshot/` however because the `xcodebuild` and in turn the `tee` command gets run multiple times, the log file is overwritten each time the tests run. This means that when the last run succeeds and the failure is before that, I have no way of working out what is going on :/. 

By having a unique filename for each `xcodebuild` command performed, i'm able to look back through the raw logs once `fastlane snapshot` has completed running and the important data wouldn't have been overwritten. This is also favoured over the original approach of appending `-a` to the `tee` command because that approach didn't have any way of keeping the log sizes reasonable.

I've tested this in a sample iOS project with a Snapfile that looks like the following:

```
# A list of devices you want to take the screenshots from
 devices([
   "iPhone 6",
   "iPhone SE"
 ])

languages([
  "en-US",
  ["pt", "pt_BR"] # Portuguese with Brazilian locale
])

# The name of the scheme which contains the UI Tests
 scheme "Example"
```

There are also a couple of extra tests to make sure things are working in all scenarios

-----

<details><summary>Original PR</summary>

### Description
Added the `-a` flag to the `tee` command in order to append the xcodebuild logs to the existing file when running snapshot in the `test_command_generator.rb`.

### Motivation and Context
I am currently running snapshot on an array of devices however my ui tests are failing randomly in some runs but not always the final device run. I try to look at the console output from xcpretty however it's not actually giving me anything useful. 

Instead I've tried looking at the output in `~/Library/Logs/snapshot/` however because the `xcodebuild` and in turn the `tee` command gets run multiple times, the log file is overwritten each time the tests run. This means that when the last run succeeds and the failure is before that, I have no way of working out what is going on :/. 

Adding the `-a` argument to `tee` means that the stream is appended to the end of the file rather than overwriting it each time so this will fix my issue and allow me to see the complete logs and debug further. 

I do have slight concerns about the size of the logs if it is always appending? Would this be an issue? I don't know how cleanup operations work here? I guess another option could be to append the device name to `xcodebuild_log_path` so that there are multiple files created instead however my Ruby knowledge doesn't span much further than being able to add an extra three characters into a string unfortunately. 

In terms of how I have tested it, I haven't yet. I'm not sure how to setup my machine to run from source code so if anybody can help that would be great (ruby/gems aren't my strong point). Thanks.
</details>

Taken from https://github.com/fastlane/fastlane/pull/8725 via @liamnichols

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
Snapshot Snapshot::TestCommandGenerator Valid iOS Configuration default options uses the default parameters | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator Valid iOS Configuration default options allows to supply custom xcargs | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator Valid iOS Configuration default options uses the default parameters on tvOS too | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator Valid macOS Configuration uses default parameters on macOS | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator Unique logs uses correct name and language | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|
Snapshot Snapshot::TestCommandGenerator Unique logs uses includes locale if specified | <a href='https://github.com/fastlane/fastlane/blob/7bd0d8c8fbf86ad1ce6aac734751dc333f5a1a17/./snapshot/spec/test_command_generator_spec.rb'>test_command_generator_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Review by ohayon — APPROVED** [review]
This is great! 💯  🚀

**Comment by KrauseFx** [comment]
Just waiting for @liamnichols's approval for the Google CLA before merging 👍

**Comment by liamnichols** [comment]
Oh cool this looks great! Thanks for picking the rest of it up 😬.  

I'm happy with the CLA stuff! (is that enough for the google bot? 🤔)

**Comment by KrauseFx** [comment]
Beautiful, thanks @liamnichols for contributing to fastlane 👍

**Pull request merged**
